community/kernel/src/test/java/org/neo4j/kernel/impl/store/MetaDataStoreTest.java

```diff
@@ -399,22 +391,33 @@ public void testRecordTransactionClosed() throws Exception
     }
 
     @Test
-    public void setUpgradeTransactionMustBeAtomic() throws Exception
+    public void setUpgradeTransactionMustBeAtomic() throws Throwable
     {
         try ( MetaDataStore store = newMetaDataStore() )
         {
             PagedFile pf = store.storeFile;
             store.setUpgradeTransaction( 0, 0, 0 );
-            CountDownLatch runLatch = new CountDownLatch( 1 );
-            AtomicBoolean stopped = new AtomicBoolean();
-            AtomicLong counter = new AtomicLong();
-
-            Runnable writer = untilStopped( stopped, runLatch, () -> {
-                long count = counter.incrementAndGet();
+            AtomicLong writeCount = new AtomicLong();
+            AtomicLong fileReadCount = new AtomicLong();
+            AtomicLong apiReadCount = new AtomicLong();
+            int limit = 10_000;
+            long endTime = currentTimeMillis() + SECONDS.toMillis( 1 );
```

Comment:
with that small endTime it's really big chance that on CI it will just will not continue based on endTime, i think we need to wait for at least of couple writes to happen